services/payments/secrets.go

```diff
@@ -106,6 +120,119 @@ func nitroAwsCfg(ctx context.Context) (aws.Config, error) {
 
 var errSecretsNotLoaded = errors.New("secrets are not yet loaded")
 
+func (s *Service) createVault(
+	ctx context.Context,
+	request paymentLib.CreateVaultRequest,
+	approverKey string,
+) (*paymentLib.CreateVaultResponse, error) {
+	vault, err := vaultFromRequest(ctx, request.Operators, request.Threshold, approverKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate vault from request: %w", err)
+	}
+	err = s.datastore.InsertVault(ctx, *vault)
+	if err != nil {
+		return nil, fmt.Errorf("failed to insert vault into QLDB: %w", err)
+	}
+
+	return &vault.shares, nil
+}
+
+func (s *Service) verifyVault(
+	ctx context.Context,
+	request paymentLib.VerifyVaultRequest,
+	approverKey string,
+) (*paymentLib.VerifyVaultResponse, error) {
+	fetchedVault, err := s.datastore.GetVault(
+		ctx,
+		request.PublicKey,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to insert vault into QLDB: %w", err)
+	}
+
+	return &paymentLib.VerifyVaultResponse{
+		Operators: fetchedVault.OperatorKeys,
+		Threshold: fetchedVault.Threshold,
+		PublicKey: fetchedVault.PublicKey,
+	}, nil
+}
+
+func vaultFromRequest(
```

Comment:
Some of what's happening in this function could be extracted in smaller functions, which would help overall, and with testing.

Reply:
We ended up doing this during a refactor so this one should be addressed